packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -830,7 +830,7 @@ class RenderShaderMask extends RenderProxyBox {
   void paint(PaintingContext context, Offset offset) {
     if (child != null) {
       assert(needsCompositing);
-      final Rect rect = Offset.zero & size;
+      final Rect rect = offset & size;
```

Comment:
The offset is passed to pushShaderMask so it can be passed onward to the painter() callback.

pushClipRect() internally applies the offset to the clipRect and uses the result to construct the ClipRectLayer.

We could do something similar here - call _shaderCallback(offset & size) in RenderShaderMask.paint to build the shader.  Then, in pushShaderMask, use maskRect.shift(offset) as the maskRect when constructing the ShaderMaskLayer.

Would that be more consistent?